src/plugins/intel_gpu/src/graph/impls/ocl/convolution.cpp

```diff
@@ -31,13 +31,19 @@ struct convolution_impl : typed_primitive_impl_ocl<convolution> {
 
         auto outer_id = _outer.id();
         auto data_type = instance.node.input().get_output_layout().data_type;
+        cldnn::data_types weights_data_type;
+        if (instance.get_network().is_internal()) {
+           weights_data_type = instance.node.weights().get_output_layout().data_type;
+        } else {
+           weights_data_type = instance.weights_memory(0)->get_layout().data_type;
+        }
 
         // Integer signed/unsigned is ok for convoluiton
```

Comment:
please create unit tests for that case

Reply:
@vladimir-paramuzov So unit test for getting instance.node.weights().get_output_layout().data_type? Do we have similar test case in clDNN unit test?

Reply:
@vladimir-paramuzov Convolution unit test already includes the case of weights located in input_layout primitive. https://github.com/wilson-seok/openvino/blob/749f15cf8b7c9db5a50b5bda4c458f883412e529/src/plugins/intel_gpu/tests/test_cases/convolution_gpu_test.cpp#L1298
Do you mean this kind of test or another one?

Reply:
Yes, something like that, but new test case should cover fixed scenario, i.e. it must fail w/o your fix.

Reply:
@vladimir-paramuzov Added unit test for the case. Thanks for the comments!